dashboard/app/controllers/reports_controller.rb

```diff
@@ -56,6 +56,25 @@ def all_usage
     render 'usage', formats: [:html]
   end
 
+  def search_for_teachers
+    authorize! :read, :reports
```

Comment:
yes, this is requiring admin access. You can write a unit test, there should be some examples in ReportsControllerTest